sklearn/discriminant_analysis.py

```diff
@@ -443,11 +446,24 @@ def fit(self, X, y):
             self.priors_ = self.priors_ / self.priors_.sum()
 
         # Get the maximum number of components
+
+        # Maximum number of components no matter what n_components is
+        # specified:
+        max_components = min(len(self.classes_) - 1, X.shape[1])
+
         if self.n_components is None:
-            self._max_components = len(self.classes_) - 1
+            self._max_components = max_components
         else:
-            self._max_components = min(len(self.classes_) - 1,
-                                       self.n_components)
+            if self.n_components > max_components:
+                warnings.warn(
+                    "n_components cannot be superior to min(n_features, "
```

Comment:
I would say "larger than" not "superior". I'm not a native English speaker and the usage strikes me as odd. Even if it's correct, we have many users that are not native English speakers and will be thrown off by it.